src/releasemonitor.cpp

```diff
@@ -39,7 +39,10 @@ void ReleaseMonitor::runInternal() {
           &ReleaseMonitor::updateRequired);
   connect(updater, &Updater::updateRecommended, this,
           &ReleaseMonitor::updateRecommended);
-  connect(updater, &QObject::destroyed, this, &ReleaseMonitor::releaseChecked);
+  connect(updater, &QObject::destroyed, [this]() {
+    emit updateCheckResult(false);
```

Comment:
in this way, when the object is released, we go back to `updateCheckResult = false` and we will open the no-update popup :/

Reply:
I guess you need to change the `Updater` class and expose a boolean flag `updateRequiredOrRecommended()` to be called here.